matrix_client/client.py

```diff
@@ -287,6 +287,9 @@ def _process_state_event(self, state_event, current_room):
             current_room.topic = state_event["content"].get("topic", None)
         elif etype == "m.room.aliases":
             current_room.aliases = state_event["content"].get("aliases", None)
+        elif etype == "m.room.member":
+            member_id = state_event["user_id"]
+            current_room.members[member_id] = User(self.api, member_id)
```

Comment:
The only thing I would say here is a "m.room.member" event covers joins,invites,leaving,bans and even profile updates. I would suggest you check state_event["content"]["membership"] == 'join' just to be sure.